scripts/Crawlers/BaseCrawler.py

```diff
@@ -0,0 +1,419 @@
+            description["keywords"]: Tags associated with the dataset, which will help in its discovery
+            description["types"]: A term, ideally from a controlled terminology, identifying the dataset type or nature of the data, placing it in a typology
+        More fields can be added as long as they comply with the DATS schema available at
+        https://github.com/CONP-PCNO/schema/blob/master/dataset_schema.json
+        
+        Any fields/keys not in the schema will be ignored when creating the dataset's DATS.
+        It is fine to add more helpful information for other methods which will use them.
+        
+        Here are some examples of valid DATS.json files:
+        https://github.com/conp-bot/conp-dataset-Learning_Naturalistic_Structure__Processed_fMRI_dataset/blob/476a1ee3c4df59aca471499b2e492a65bd389a88/DATS.json
+        https://github.com/conp-bot/conp-dataset-MRI_and_unbiased_averages_of_wild_muskrats__Ondatra_zibethicus__and_red_squirrels__Tami/blob/c9e9683fbfec71f44a5fc3576515011f6cd024fe/DATS.json
+        https://github.com/conp-bot/conp-dataset-PERFORM_Dataset__one_control_subject/blob/0b1e271fb4dcc03f9d15f694cc3dfae5c7c2d358/DATS.json
+
+        Returns:
+            List of description of relevant datasets. Each description is a
+            dictionary. For example:
+
+            [{
+                "title": "PERFORM Dataset Example",
+                "description": "PERFORM dataset description",
+                "version": "0.0.1",
+                ...
+             },
+             {
+                "title": "SIMON dataset example",
+                "description: "SIMON dataset description",
+                "version": "1.4.2",
+                ...
+             },
+             ...
+            ]
+        """
+        return []
+
+    @abc.abstractmethod
+    def add_new_dataset(self, dataset_description, dataset_dir):
+        """
+        Configure and add newly created dataset to the local CONP git repository.
+
+        This is where newly crawled dataset which are not present locally are
+        created. This is where a one time configuration such as modifying
+        .gitignore, .gitattribute or no-annex some files can be done.
+        It is possible to add a tracker file in the repo in order
+        to track in future runs if there are any changes to the dataset.
+        For example, you can store a dataset version in '.version' file
+        so that if the dataset version changes on the remote platform,
+        update_if_necessary() will detect it and update the dataset.
+
+        Parameter:
+        dataset_description (dict): Dictionary containing information on
+                                    retrieved dataset from platform. Element of 
+                                    the list returned by get_all_dataset_description
+        dataset_dir (str): Local directory path where the newly
+                           created datalad dataset is located. 
```

Comment:
Can you clarify if the dataset dir is created and initialized by the base class, and explain what the crawler has to do? It looks like the crawler only has to download files there, does it also have to add them to the datalad repo? If not, how are archives handled?